lib/tests/tests.c

```diff
@@ -3117,6 +3117,84 @@ compute_beta_coalescence_rate_fails(unsigned int num_ancestors, double alpha)
     return 0;
 }
 
+static void
+check_time_change(msp_t *msp, double t)
+{
+    const double tol = 1e-9;
+    double g, g2, t2;
+    g = msp->model.model_time_to_generations(&msp->model, t);
+    t2 = msp->model.generations_to_model_time(&msp->model, g);
+    CU_ASSERT_DOUBLE_EQUAL(t, t2, tol);
+    g2 = msp->model.generation_rate_to_model_rate(&msp->model, t);
+    CU_ASSERT_DOUBLE_EQUAL(g, g2, tol);
+}
+
+static void
+test_model_time_change_consistency1(double population_size, double t)
+{
+    int ret;
+    msp_t msp;
+    const unsigned int n = 10;
+    const double alpha = 1.5;
+    const double truncation_point = 1.0;
+    const double psi = 0.5;
+    const double c = 5.0;
+    sample_t *samples = malloc(n * sizeof(sample_t));
+    gsl_rng *rng = gsl_rng_alloc(gsl_rng_default);
+    recomb_map_t recomb_map;
+    tsk_table_collection_t tables;
+
+    ret = recomb_map_alloc_uniform(&recomb_map, 1.0, 1, true);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_init(&tables, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_FATAL(samples != NULL);
+    CU_ASSERT_FATAL(rng != NULL);
+    memset(samples, 0, n * sizeof(*samples));
+    ret = msp_alloc(&msp, n, samples, &recomb_map, &tables, rng);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+
+    ret = msp_set_simulation_model_hudson(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_smc(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_smc_prime(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_dtwf(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_wf_ped(&msp, population_size);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_dirac(&msp, population_size, psi, c);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+    ret = msp_set_simulation_model_beta(&msp, population_size, alpha, truncation_point);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    check_time_change(&msp, t);
+
+    msp_free(&msp);
+    recomb_map_free(&recomb_map);
+    tsk_table_collection_free(&tables);
+    free(samples);
+    gsl_rng_free(rng);
+}
+
+static void
+test_model_time_change_consistency(void)
+{
+    double N;
+    for (N=1.0; N<1e9; N+=N) {
```

Comment:
This is a **lot** of Ns! Something like ``double Ns = {1e-6, 1e-1, 1, 2, 3, 10, 1e4, 1e9}`` would probably cover all the interesting cases I'd say.